### PR TITLE
Wizard Kafka Connect form

### PR DESCRIPTION
**Pull request by David-DB88**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
created Kafka Connect form
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  remove,
+  fields,
+}) => {
+  const [newKafkaConnect, setNewKafkaConnect] = useState(false);
+  const methods = useFormContext();
+  const connect = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.preventDefault();
+    setNewKafkaConnect(!newKafkaConnect);
+  };
+  const kafka = methods.getValues('kafkaConnect');
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:36]:
```suggestion
  const connects = methods.getValues('kafkaConnect');
```

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import PlusIcon from 'components/common/Icons/PlusIcon';
+import IconButtonWrapper from 'components/common/Icons/IconButtonWrapper';
+import CloseIcon from 'components/common/Icons/CloseIcon';
+
+type PropType = {
+  handleAddKafkaConnect: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  remove: (index: number) => void;
+  fields: TField[];
+};
+type TField = {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:17]: We are not using such prefixes as `T` or `I` for types and interfaces in our code base. Please use something more descriptive, like `KafkaConnectFormProps`

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+import { ErrorMessage } from '@hookform/error-message';
+import { useFormContext } from 'react-hook-form';
+import PlusIcon from 'components/common/Icons/PlusIcon';
+import IconButtonWrapper from 'components/common/Icons/IconButtonWrapper';
+import CloseIcon from 'components/common/Icons/CloseIcon';
+
+type PropType = {
+  handleAddKafkaConnect: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  remove: (index: number) => void;
+  fields: TField[];
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:15]: It is not a fields. It is a kafkaConnects

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+import { FormError } from 'components/common/Input/Input.styled';
+import { ErrorMessage } from '@hookform/error-message';
+import { useFormContext } from 'react-hook-form';
+import PlusIcon from 'components/common/Icons/PlusIcon';
+import IconButtonWrapper from 'components/common/Icons/IconButtonWrapper';
+import CloseIcon from 'components/common/Icons/CloseIcon';
+
+type PropType = {
+  handleAddKafkaConnect: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  remove: (index: number) => void;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:14]: What's wrong with naming? You have `handleAddKafkaConnect` and just `remove`

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  password: string;
+};
+const KafkaConnect: React.FC<PropType> = ({
+  handleAddKafkaConnect,
+  remove,
+  fields,
+}) => {
+  const [newKafkaConnect, setNewKafkaConnect] = useState(false);
+  const methods = useFormContext();
+  const connect = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:32]: what is this connect means? Are you connecting to some remote server or what?

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx (hunk 1 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:36]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx (hunk 2 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:17]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx (hunk 3 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:15]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx (hunk 4 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:14]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx (hunk 5 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:32]: just showing  connects, i will replace with showConnects

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+              buttonSize="M"
+              buttonType="secondary"
+              onClick={() =>
+                append({
+                  name: '',
+                  url: '',
+                  isAuth: false,
+                  username: '',
+                  password: ',',
+                })
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:101]: create handler function

File: kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.styled.ts [hunk 7]
```diff
@@ -89,3 +90,11 @@ export const FlexRow = styled.div`
 export const FlexGrow1 = styled.div`
   flex-grow: 1;
 `;
+// Kafka Connect
+export const ConnectInputWrapper = styled(InputContainer)`
+  grid-template-columns: 1fr 30px;
+  max-width: 100%;
+  & > div > div {
+    padding-top: 1rem;
+  }
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.styled.ts:99]: just create styled component for thgis nested div

File: kafka-ui-react-app/src/components/common/Checkbox/Checkbox.tsx [hunk 8]
```diff
@@ -8,14 +8,15 @@ interface CheckboxProps {
...
 }
 
-const Checkbox: React.FC<CheckboxProps> = ({ name, label, hint }) => {
+const Checkbox: React.FC<CheckboxProps> = ({ name, label, hint, cursor }) => {
   const { register } = useFormContext();
 
   return (
     <div>
-      <InputLabel>
+      <InputLabel cursor={cursor}>
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Checkbox/Checkbox.tsx:19]: Label should not have pointer cursor

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+  url: string;
+  isAuth: boolean;
+  username: string;
+  password: string;
+};
+const KafkaConnect: React.FC<PropType> = ({
+  addConnect,
+  removeConnect,
+  kafkaConnects,
+}) => {
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx]: ок

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+  const methods = useFormContext();
+  const showConnects = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.preventDefault();
+    setNewKafkaConnect(!newKafkaConnect);
+  };
+  const connects = methods.getValues('kafkaConnect');
+  useEffect(() => {
+    if (connects.length < 1) {
+      setNewKafkaConnect(false);
+      methods.reset({
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx]: ок

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+            username: '',
+            password: ',',
+          },
+        ],
+      });
+    }
+  }, [connects]);
+  return (
+    <S.Section>
+      <S.SectionName>Kafka Connect</S.SectionName>
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx]: ок

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+          <S.KafkaConnect>
+            {kafkaConnects.map((item, index) => (
+              <div key={item.id}>
+                <S.InputsContainer>
+                  <div>
+                    <S.PartStyled>
+                      <S.ItemLabelRequired>
+                        <label htmlFor="kafkaConnect.name">
+                          Kafka Connect name
+                        </label>{' '}
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx]: ок

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+    name: 'kafkaConnect',
+  });
+  const connects = getValues('kafkaConnect');
+  useEffect(() => {
+    if (connects.length < 1) {
+      setNewKafkaConnect(false);
+      reset({
+        ...getValues(),
+        kafkaConnect: [],
+      });
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:30]: ok

File: kafka-ui-react-app/src/components/common/Checkbox/Checkbox.tsx (hunk 8 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Checkbox/Checkbox.tsx:19]: Roma told me several times that I need to make on label cursor  pointer

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+import CloseIcon from 'components/common/Icons/CloseIcon';
+import Heading from 'components/common/heading/Heading.styled';
+import Checkbox from 'components/common/Checkbox/Checkbox';
+import {
+  FlexGrow1,
+  FlexRow,
+} from 'components/Wizard/WizardForm/WizardForm.styled';
+
+const KafkaConnect = () => {
+  const [newKafkaConnect, setNewKafkaConnect] = useState(false);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaConnect/KafkaConnect.tsx:17]: why it is false by default? What if we already have connects in the list

**Review by workshur — APPROVED** [review]

**Pull request merged**